src/mod_fastcgi.c

```diff
@@ -3552,6 +3593,27 @@ static handler_t fcgi_check_extension(server *srv, connection *con, void *p_d, i
 
 			if (buffer_is_empty(ext->key)) continue;
 
+			/* do not reuse same authorizer, skip if it's in FastCGI-Authorizer env var */
```

Comment:
What is the use case here?  con->loops_per_request test elsewhere will prevent infinite loops.
Should the 'continue' be 'break'?  Should the check be done once below the loop, only if there is a match?

Also, the implementation below only be a few lines of code, but it could probably be more efficient if you also modify the code that inserts FastCGI-Authorizer into the environment.  If you start FastCGI-Authorizer with '|' and modify the existing value and always append 'ext|', then you can strstr for '|ext|'.

BTW, I am still mulling over the choice of 'FastCGI-Authorizer' env var name.  It will leak into CGI, FastCGI, SCGI, and SSI environments, since lighttpd does not have the equivalent to Apache module 'notes', which do not leak into subprocess environments.